### ISSUE TITLE
Certificate resource status can't be checked with current pass/fail conditions.

### ISSUE THREAD

**Issue by rendhalver** [Shopify/kubernetes-deploy#472]
# Bug report

Certificate resource status can't be checked with current pass/fail conditions.

**Expected behavior:** [What you expected to happen]

Adding annotations to the Certificate CRD would allow us to monitor pass/fail conditions during a deploy.

**Actual behavior:** [What actually happened]

We were attempting to add annotations to the CRDs responsible for `Certificate` resources so  kubernetes-deploy could test for whether those resources were successfully deployed.
Currently all resources deployed with CRD's are expected to provide an `ObservedGeneration` status message.
the problem is the cert-manager controller doesn't add those statuses so we are unable to use the pass/fail logic in kubernetes-deploy to check those resources.

We opened an issue on cert-manager to ask about getting `ObservedGeneration` status added
to the CRD. Please see there for the discussion.
The short version is do you expect every CRD provider to provide an ObservedGeneration status message so it's pass/fail conditions can be monitored during a deploy.
https://github.com/jetstack/cert-manager/issues/1551

**Version(s) affected:** [run `kubernetes-deploy --version`]
v0.25.0

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Comment by dturn** [comment]
> The short version is do you expect every CRD provider to provide an ObservedGeneration status message so it's pass/fail conditions can be monitored during a deploy.

Yes, we do.

These checks exist to prevent a race conditions that we slammed into repeatedly where we examined the resource and declared success/failure based on old state. I'm sure we could find a way to relax the constraint, but it would mean the results would occasionally be wrong and we don't think that's the right trade off.

**Comment by benlangfeld** [comment]
The cert-manager project has declined to implement observedGeneration. Is there anything that we could do in Kubernetes-deploy to help support controllers like this?

**Comment by dturn** [comment]
We still think its' reasonable for cert manager to update observedGeneration and since we can't give a reliable result without it we don't see any other path forward. Feel free to re-open if you suggestions we might have missed.
